repose-aggregator/components/filters/keystone-v2-filter/src/main/scala/org/openrepose/filters/keystonev2/AbstractKeystoneV2Filter.scala

```diff
@@ -0,0 +1,146 @@
+/*
+ * _=_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_=
+ * Repose
+ * _-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-
+ * Copyright (C) 2010 - 2015 Rackspace US, Inc.
+ * _-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * =_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_=_
+ */
+package org.openrepose.filters.keystonev2
+
+import javax.servlet._
+import javax.servlet.http.HttpServletResponse._
+import javax.servlet.http.{HttpServletRequest, HttpServletResponse}
+
+import com.rackspace.httpdelegation.HttpDelegationManager
+import com.typesafe.scalalogging.slf4j.LazyLogging
+import org.openrepose.commons.utils.http.{IdentityStatus, OpenStackServiceHeader}
+import org.openrepose.commons.utils.servlet.http.ResponseMode.{MUTABLE, PASSTHROUGH}
+import org.openrepose.commons.utils.servlet.http.{HttpServletRequestWrapper, HttpServletResponseWrapper}
+import org.openrepose.core.filter.AbstractConfiguredFilter
+import org.openrepose.core.services.config.ConfigurationService
+import org.openrepose.filters.keystonev2.config._
+
+import scala.collection.JavaConverters._
+import scala.reflect.ClassTag
+import scala.util.{Failure, Success, Try}
+
+abstract class AbstractKeystoneV2Filter[T <: KeystoneV2Config: ClassTag](configurationService: ConfigurationService)
+  extends AbstractConfiguredFilter[T](configurationService)
+    with HttpDelegationManager
+    with LazyLogging {
+
+  import AbstractKeystoneV2Filter._
+
+  val handleFailures: PartialFunction[Try[Unit.type], KeystoneV2Result]
+
+  def doAuth(request: HttpServletRequestWrapper): Try[Unit.type]
+
+  def doWork(servletRequest: HttpServletRequest, servletResponse: HttpServletResponse, chain: FilterChain): Unit = {
+    /**
+      * STATIC REFERENCE TO CONFIG
+      */
+    val config = configuration
```

Comment:
Since we're using the `configuration` member object in the concrete implementation, I don't see much value in keeping this around here, either.

Reply:
Oh, and if we do get rid of these references, do we want to pop all of the nested methods out? I personally vote yes.

Reply:
I get the desire, but it's adding work onto an already long running story.